src/simulations/collect.jl

```diff
@@ -343,19 +345,29 @@ function multi_agent_agg_types!(types::Vector{Vector{T} where T}, utypes::Tuple,
 end
 
 """
-    aggname(k) → name
-    aggname(k, agg) → name
-    aggname(k, agg, condition) → name
+    dataname(k) → name
+    dataname(k, agg) → name
+    dataname(k, agg, condition) → name
 
 Return the name of the column of the `i`-th collected data where `k = adata[i]`
 (or `mdata[i]`).
-`aggname` also accepts tuples with aggregate and conditional values.
+`dataname` also accepts tuples with aggregate and conditional values.
 """
-aggname(k, agg) = string(agg)*"_"*string(k)
-aggname(k, agg, condition) = string(agg)*"_"*string(k)*"_"*string(condition)
-aggname(x::Tuple{K,A}) where {K,A} = aggname(x[1], x[2])
-aggname(x::Tuple{K,A,C}) where {K,A,C} = aggname(x[1], x[2], x[3])
-aggname(x::Union{Function, Symbol, String}) = string(x)
+dataname(k, agg) = string(agg)*"_"*string(k)
+@deprecate aggname(k, agg) dataname(k, agg)
+dataname(k, agg, condition) = string(agg)*"_"*string(k)*"_"*string(condition)
+@deprecate aggname(k, agg, condition) dataname(k, agg, condition)
+dataname(x::Tuple{K,A}) where {K,A} = dataname(x[1], x[2])
+@deprecate (aggname(x::Tuple{K,A}) where {K,A}) dataname(x)
+dataname(x::Tuple{K,A,C}) where {K,A,C} = dataname(x[1], x[2], x[3])
+@deprecate (aggname(x::Tuple{K,A,C}) where {K,A,C}) dataname(x)
+dataname(x::Union{Symbol,String}) = string(x)
+@deprecate aggname(x::Union{Symbol,String}) dataname(x)
+# This takes care to include fieldnames and values in the column name to make column names unique
+# if the same function is used with different values of outer scope variables.
+dataname(x::Function) = join(
+    vcat([string(x)], ["$(prop)=$(getproperty(x, prop))" for prop in propertynames(x)]), "_")
+@deprecate aggname(x::Function) dataname(x)
```

Comment:
A single `@deprecate aggname dataname` is enough, no reason to repeat it everywhere.

Reply:
Ok, thanks! I thought it needed mapping for every method of `dataname`/`aggname`. I'm new to Julia's multiple dispatch infrastructure.

Reply:
No problem. multiple dispatch is the best thing in the programming universe in my opinion :D

Reply:
I've replaced the multiple `@deprecate` lines with a single `@deprecate aggname dataname` in the latest push.

Reply:
> No problem. multiple dispatch is the best thing in the programming universe in my opinion :D

Ah, I agree!! I fell in love with that concept once I grasped it!